### ISSUE TITLE
Can not implement custom runner

### ISSUE THREAD

**Issue by kuntoro-adi** [open-mmlab/mmsegmentation#1419]
Hi,
I want to implement my custom runner. I have done the following

1. Create directory "runner" under "mmseg"

2. Include `__init__.py` as the following
```
from .builder import RUNNERS, build_runner
from .custom_runner import CustomRunner

__all__ = [
    'build_runner', 'RUNNERS', 'CustomRunner'
]
```

3. Include `builder.py` (copied from mmcv source code https://github.com/open-mmlab/mmcv/)
```
# Copyright (c) OpenMMLab. All rights reserved.
import copy

from mmcv.utils import Registry

RUNNERS = Registry('runner')
RUNNER_BUILDERS = Registry('runner builder')


def build_runner_constructor(cfg):
    return RUNNER_BUILDERS.build(cfg)


def build_runner(cfg, default_args=None):
    runner_cfg = copy.deepcopy(cfg)
    constructor_type = runner_cfg.pop('constructor',
                                      'DefaultRunnerConstructor')
    runner_constructor = build_runner_constructor(
        dict(
            type=constructor_type,
            runner_cfg=runner_cfg,
            default_args=default_args))
    runner = runner_constructor()
    return runner
```

4. Include `custom_dataset.py` (copied from IterBasedRunner by mmcv)
```
import os.path as osp
import platform
import shutil
import time
import warnings

import torch
from torch.optim import Optimizer

import mmcv
from mmcv.runner.base_runner import BaseRunner
from mmcv.runner import RUNNERS
from mmcv.runner.checkpoint import save_checkpoint
from mmcv.runner.hooks import IterTimerHook
from mmcv.runner.utils import get_host_info


class IterLoader:

    def __init__(self, dataloader):
        self._dataloader = dataloader
        self.iter_loader = iter(self._dataloader)
        self._epoch = 0

    @property
    def epoch(self):
        return self._epoch

    def __next__(self):
        try:
            data = next(self.iter_loader)
        except StopIteration:
            self._epoch += 1
            if hasattr(self._dataloader.sampler, 'set_epoch'):
                self._dataloader.sampler.set_epoch(self._epoch)
            time.sleep(2)  # Prevent possible deadlock during epoch transition
            self.iter_loader = iter(self._dataloader)
            data = next(self.iter_loader)

        return data

    def __len__(self):
        return len(self._dataloader)


@RUNNERS.register_module()
class CustomRunner(BaseRunner):

    def train(self, data_loader, **kwargs):
        self.model.train()
        self.mode = 'train'
        self.data_loader = data_loader
        self._epoch = data_loader.epoch
        data_batch = next(data_loader)
        self.call_hook('before_train_iter')
        outputs = self.model.train_step(data_batch, self.optimizer, **kwargs)
        if not isinstance(outputs, dict):
            raise TypeError('model.train_step() must return a dict')
        if 'log_vars' in outputs:
            self.log_buffer.update(outputs['log_vars'], outputs['num_samples'])
        self.outputs = outputs
        self.call_hook('after_train_iter')
        self._inner_iter += 1
        self._iter += 1

    ...
```

However, I got this error
```
KeyError: 'CustomRunner is not in the runner registry'
```

How to fix this?

Thank you very much.

**Comment by MengzhangLI** [comment]
You mean like custome imports?

https://github.com/open-mmlab/mmdetection/blob/5e246d5e3bc3310b5c625fb57bc03d2338ca39bc/docs/en/tutorials/customize_runtime.md

**Comment by kuntoro-adi** [comment]
Yes. 
I moved the "runner" directory into "mmseg/core".
I think the problem is that there are two `__init__.py` that must be configured. The one is inside the "runner" directory and the other one is inside the "core". I got error when only the `__init__.py` inside "runner" was configured. In my case, the following should be added to the `__init__.py` inside the "core"
```
from .runner import *
```
The code works now. Thank you.
